lib/matplotlib/testing/conftest.py

```diff
@@ -125,3 +125,46 @@ def test_imshow_xarray(xr):
 
     xr = pytest.importorskip('xarray')
     return xr
+
+
+@pytest.fixture
+def _text_placeholders(monkeypatch):
+    from matplotlib.patches import Rectangle
+
+    def patched_get_text_metrics_with_cache(renderer, text, fontprop, ismath, dpi):
+        """
+        Replace ``_get_text_metrics_with_cache`` with fixed results.
+
+        The usual ``renderer.get_text_width_height_descent`` would depend on font
+        metrics; instead the fixed results are based on font size and the length of the
+        string only.
+        """
+        # While get_window_extent returns pixels and font size is in points, font size
+        # includes ascenders and descenders. Leaving out this factor and setting
+        # descent=0 ends up with a box that is relatively close to DejaVu Sans.
+        height = fontprop.get_size()
+        width = len(text) * height / 1.618  # Golden ratio for character size.
+        descent = 0
+        return width, height, descent
+
+    def patched_text_draw(self, renderer):
+        """
+        Replace ``Text.draw`` with a fixed bounding box Rectangle.
+
+        The bounding box corresponds to ``Text.get_window_extent``, which ultimately
+        depends on the above patched ``_get_text_metrics_with_cache``.
+        """
+        if renderer is not None:
+            self._renderer = renderer
+        if not self.get_visible():
+            return
+        if self.get_text() == '':
+            return
+        bbox = self.get_window_extent()
+        rect = Rectangle(bbox.p0, bbox.width, bbox.height,
+                         facecolor=(self.get_color(), 0.5), edgecolor='none')
```

Comment:
I'm a bit torn on using transparency here. On the one hand, that correctly catches overlay (but that should be rare). On the other hand, alpha blending is an additional component determining the output image, can that lead to more unstable reference images in the future?

Reply:
Test images looks a bit redacted now 😆, but I think this is better.